include/arch/riscv/arch/object/structures.h

```diff
@@ -145,6 +145,23 @@ static inline void *CONST cap_get_archCapPtr(cap_t cap)
 
 static inline bool_t CONST Arch_isCapRevocable(cap_t derivedCap, cap_t srcCap)
 {
+#ifdef CONFIG_ALLOW_SBI_CALLS
+    switch (cap_get_capType(derivedCap)) {
+    case cap_sbi_cap:
+        if (!cap_sbi_cap_get_capSBIEIDBadged(derivedCap) || !cap_sbi_cap_get_capSBIFIDBadged(derivedCap)) {
+            return true;
+        }
+
+        return (cap_sbi_cap_get_capSBIFIDBadge(derivedCap) !=
+            cap_sbi_cap_get_capSBIFIDBadge(srcCap) ||
+            cap_sbi_cap_get_capSBIEIDBadge(derivedCap) !=
+            cap_sbi_cap_get_capSBIEIDBadge(srcCap)
+            );
+
+    default:
+        return false;
+    }
+#endif
     return false;
```

Comment:
Nicer to do it the Arm way and always have the switch, no final return and the ifdef per case.